src/main/java/com/google/api/generator/gapic/composer/ServiceSettingsClassComposer.java

```diff
@@ -119,11 +120,20 @@ public GapicClass generate(Service service, Map<String, Message> ignore) {
     return GapicClass.create(kind, classDef);
   }
 
-  private static List<CommentStatement> createClassHeaderComments(Service service) {
+  private static List<CommentStatement> createClassHeaderComments(
+      Service service, TypeNode classType) {
+    // Pick the first pure unary rpc method, if no such method exists, then pick the first in the
+    // list.
     Optional<Method> methodOpt =
-        service.methods().isEmpty() ? Optional.empty() : Optional.of(service.methods().get(0));
+        service.methods().isEmpty()
+            ? Optional.empty()
+            : Optional.of(
+                service.methods().stream()
+                    .filter(m -> m.stream() == Stream.NONE && !m.hasLro() && !m.isPaged())
+                    .findFirst()
+                    .orElse(service.methods().stream().findFirst().get()));
```

Comment:
Why not simply `.orElse(service.methods().get(0))`?

Reply:
sg